### PR TITLE
fix(material/slider): handle null and undefined values in slider input

### LINKED ISSUE #30614: bug(MatSlider): Inconsistent Reset Behavior for mat-slider with Range Thumbs in Reactive Forms
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

When using `mat-slider` with **range thumbs** (start and end thumbs) in a reactive form, resetting the form sets the control values to `null`. This results in:  

1. The thumbs repositioning **to slice the range into equal parts**, despite their values being `null`.  
2. A conflict with the initialization logic, where `null` values correctly position the thumbs **at the range’s endpoints**.  

##### Investigation  
- On reset, the slider attempts to **divide the range into equal sections**.  
- This behavior works fine for **single-thumb sliders**, but for **range sliders**, it introduces **rounding errors** (since the range is divided by 3, which becomes worse as the value of `steps` gets closer to `0`).  
- As a result, multiple resets can produce **different thumb positions**, making the behavior inconsistent:  
  - **Inconsistent with initialization** (where `null` places thumbs at the ends).  
  - **Inconsistent with itself** (due to rounding errors). 

### Reproduction

StackBlitz link: https://stackblitz.com/edit/components-issue-starter-ak9ciyht?file=src%2Fmain.ts
Steps to reproduce:
1. Click on `reset` button
2. Notice the thumbs are trying to slice the range into 3 parts
3. Click on `reset` button 
4. Notice the thumbs are getting into different positions

### Expected Behavior

When resetting the form, the thumbs should **match the initialization logic**:  
  - If the control values are `null`, **both thumbs should be placed at the range’s endpoints**.  
  - The reset behavior should be **consistent and predictable**.  

### Actual Behavior

Instead of moving to the endpoints, the thumbs attempt to **slice the range into equal parts**, leading to:  
  - Inconsistency with initialization (`null` values initially place thumbs at the range’s ends).  
  - Rounding errors when dividing the range, causing the reset position to change across multiple resets.  

### Environment

```
Angular CLI: 18.2.8
Node: 18.20.3
Package Manager: npm 10.2.3
OS: linux x64

Angular: 18.2.8
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, material, platform-browser, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1802.9
@angular-devkit/build-angular   18.2.9
@angular-devkit/core            18.2.9
@angular-devkit/schematics      18.2.8
@schematics/angular             18.2.8
rxjs                            7.8.1
typescript                      5.5.4
zone.js                         0.14.10
```

### PR DESCRIPTION
**Pull request by mistrykaran91**
The slider values behaved inconsistently when we reset the formGroup. This MR will resolve that issue by setting `value` to default value when its `undefined` or `null`

Fixes #30614

### CODE REVIEW

File: src/material/slider/slider.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ export class MatSlider implements AfterViewInit, OnDestroy, _MatSlider {
     return this._min;
   }
   set min(v: number) {
-    const min = isNaN(v) ? this._min : v;
+    const min = !v || isNaN(v) ? this._min : v;
```
Comment (kenyerman) [inline src/material/slider/slider.ts:143]: `!v` will be true for `0`, which won't trigger the change of `this._min`.
This can be a problem if `min` was previously set  to `-1` and then trying to set it to `0`.

Consider checking `v` against `undefined` and `null` values.

```suggestion
    const min = v === undefined || v === null || isNaN(v) ? this._min : v;
```

File: src/material/slider/slider-input.ts [hunk 2]
```diff
@@ -96,7 +96,10 @@ export class MatSliderThumb implements _MatSliderThumb, OnDestroy, ControlValueA
     return numberAttribute(this._hostElement.value, 0);
   }
   set value(value: number) {
-    value = isNaN(value) ? 0 : value;
+    value = value && isNaN(value) ? 0 : value;
+    if (value === null || value === undefined) {
+      value = this._getDefaultValue();
+    }
```
Comment (kenyerman) [inline src/material/slider/slider-input.ts:102]: i think it would be more readable if we do the `null` and `undefined` checks first and then overwrite the value if it's `NaN`. This results in a simpler diff.

```suggestion
    if (value === null || value === undefined) {
      value = this._getDefaultValue();
    }
    value = isNaN(value) ? 0 : value;
```

**Review by kenyerman — CHANGES_REQUESTED** [review]
thank you for working on this! ❤️
i've found some falsyness checks that can be improved, please review them and correct me if i'm wrong.

File: src/material/slider/slider.ts (hunk 1 above)
Comment (mistrykaran91) [reply src/material/slider/slider.ts]: Agreed, I first thought of this whether this change is required or not. But yeah I'll make the suggested changes.

File: src/material/slider/slider-input.ts (hunk 2 above)
Comment (mistrykaran91) [reply src/material/slider/slider-input.ts]: Done, however one test seems to be failing not sure why. I might have missed some edge case I think so

Comment (mistrykaran91) [reply src/material/slider/slider-input.ts]: I have removed `undefined` check and it seems test cases failed if I checked for `undefined` may be that was some edge where we want to set `undefined`. 
If not then I think we need to update test cases where we set `undefined` to `endVal` in one of the test cases.

**Review by kenyerman — APPROVED** [review]
looks good to me! 🚀

**Review by wagnermaciel — APPROVED** [review]
